### PR TITLE
fix(sdk): when in disable mode don't spin up service

### LINKED ISSUE #4529: [CLI]: setting wandb disabled doesn't fully disable it
### Describe the bug

I'm using a performance profiler `viztracer` on a pytorch training routine that uses wandb. It seems like there's some interference between the tracer and wandb that causes wandb to crash. So I figure I can just disable wandb (`$ wandb disabled`) , but it still crashes with a bug deep inside wandb - it seems like wandb is still trying to launch the service process or something even when it's set to disabled!

Here's the traceback - the first 2 lines are from some logging that I added in the wandb code to try to figure out what was going on. I'm guessing the tracer somehow patched `python` and it no longer supports `-m`? But the solution IMO is really that wandb should not be trying to start a process at all when disabled. 
```
$ viztracer --min_duration 10us --ignore_c_function --ignore_frozen my_script.py
['/opt/venv/bin/python3.9', '-m', 'wandb', 'service', '--port-filename', '/tmp/tmpm5groj4d/port-12604.txt', '--pid', '12604', '--debug', '--serve-sock']
No such file as -m
Traceback (most recent call last):
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 1075, in init
    wi.setup(kwargs)
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_init.py", line 165, in setup
    self._wl = wandb_setup.setup()
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_setup.py", line 312, in setup
    ret = _setup(settings=settings)
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_setup.py", line 307, in _setup
    wl = _WandbSetup(settings=settings)
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_setup.py", line 293, in __init__
    _WandbSetup._instance = _WandbSetup__WandbSetup(settings=settings, pid=pid)
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_setup.py", line 106, in __init__
    self._setup()
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_setup.py", line 234, in _setup
    self._setup_manager()
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_setup.py", line 265, in _setup_manager
    self._manager = wandb_manager._Manager(
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/wandb_manager.py", line 108, in __init__
    self._service.start()
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/service/service.py", line 110, in start
    self._launch_server()
  File "/opt/venv/lib/python3.9/site-packages/wandb/sdk/service/service.py", line 106, in _launch_server
    assert ports_found
AssertionError
```

Btw this script is sufficient to replicate the issue when called by viztracer as shown above:
```
import wandb

if __name__ == "__main__":
    wandb.init()
```

### Additional Files

_No response_

### Environment

WandB version: 0.13.5

OS: Ubuntu 20.04

Python version: 3.9

Versions of relevant libraries:

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by kptkin**
Fixes WB-11440
Fixes #4529

Description
-----------
What does the PR do?

- removes the require_service flag and unify it with disable_service
- make disable_service a setting with a policy style of resolve order
- clean up wandb.settings tests and make them proper unit tests
- add service disable/enable to our cli (to make it backward compatible)
- allow disable mode in wandb to disable service as well (to make it backward compatible)

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/main/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/sdk/wandb_settings.py [hunk 1]
```diff
@@ -505,6 +505,11 @@ def _default_props(self) -> Dict[str, Dict[str, Any]]:
         """
         return dict(
             _disable_meta={"preprocessor": _str_as_bool},
+            _disable_service={
+                "value": False,
+                "preprocessor": _str_as_bool,
+                "is_policy": True,
```
Comment (dmitryduev) [inline wandb/sdk/wandb_settings.py:511]: Is this necessary?

**Review by dmitryduev — APPROVED** [review]
🦭 of approval approves!

File: wandb/sdk/wandb_settings.py (hunk 1 above)
Comment (dmitryduev) [reply wandb/sdk/wandb_settings.py:511]: (I mean is_policy)

Comment (kptkin) [reply wandb/sdk/wandb_settings.py:511]: I want it to retain the backward behavior. I don't want users to pass it as a setting to `wandb.run` and overwrite env variable setting

**Pull request merged**